airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/test/context.py

```diff
@@ -34,6 +35,67 @@ def __init__(
         self.run_step_options = self._skip_metadata_disabled_test_suites(self.run_step_options)
         self.step_id_to_secrets_mapping = self._get_step_id_to_secret_mapping()
 
+    @property
+    def metadata_yaml_only_change(self) -> bool:
+        """Check if the only modified file is metadata.yaml.
+
+        Returns:
+            bool: True if the only modified file is metadata.yaml, False otherwise.
+        """
+        for modified_file in self.modified_files:
+            if not str(modified_file).endswith(METADATA_FILE_NAME):
+                return False
+        return True
+
+    @property
+    def test_only_change(self) -> bool:
+        """Check if the only modified files are in unit_tests or integration_tests directories.
+
+        Returns:
+            bool: True if all modified files are in unit_tests or integration_tests, False otherwise.
+        """
+        test_directories = ("unit_tests", "integration_tests")
+        for modified_file in self.modified_files:
+            try:
+                rel_path = modified_file.relative_to(self.connector.code_directory)
+                top_level_dir = rel_path.parts[0] if rel_path.parts else ""
+                if top_level_dir not in test_directories:
+                    return False
+            except ValueError:
+                # File not under connector directory, treat as non-test change
+                return False
+        return True
+
+    @property
+    def should_run_static_analysis(self) -> bool:
```

Comment:
this method seems redundant if we instead just call the metadata_yaml_only_change and test_only_change from pipeline.py. Can we instead just delete this